analytics/analytics.go

```diff
@@ -25,6 +25,9 @@ import (
 	"github.com/pkg/errors"
 )
 
+var counts = make(map[string]map[string]int)
+var countsMutex = &sync.Mutex{}
```

Comment:
I would put this in a struct and make that struct a global singleton. (In case we need to ever make this a non-global it's easier)

Reply:
Sounds like a valid point, I moved these into a struct under a seperate analytics/counters.go file.